### ISSUE TITLE
Exception has occurred: NotImplementedError You must implement the backward function for custom autograd.Function.

### ISSUE THREAD

**Issue by LeopoldACC** [pytorch/pytorch#70175]
### 🐛 Describe the bug
@svenstaro @eklitzke @dreiss @huitseeker @jfsantos 
Hi,I'd appreciate some help. I meet a problem that when use the [code](https://github.com/POSTECH-CVLab/point-transformer/blob/master/model/pointtransformer/pointtransformer_seg.py) **to get the gradient of part input when the whole input is a `list`.**

First, I try to assign each Variable of the input list with "requires_grad=True" as shown below:
the evaluate code 
```python
def data_transorm(input):
    b,n,c = input.shape
    input = input.reshape(b*n,c)
    offset = [(i+1)*n for i in range(b)]
    return [input[:,:3].contiguous(),input[:,3].unsqueeze(-1).contiguous(),torch.IntTensor(offset)]

input_xyz,input_r,offset = data_transorm(input)
input_xyz = Variable(input_xyz,requires_grad=True)
input_r = Variable(input_r,requires_grad=True)
logits = model([input_xyz,input_r,offset])
loss = criterion(logits, target.squeeze())
loss.backward()
```
the forward code of origin code:
in which,`pxo` include `[tensor(n,3),tensor(n,c),IntTensor(b)]`
```python
def forward(self, pxo):
        p0, x0, o0 = pxo  # (n, 3), (n, c), (b)
        x0 = p0 if self.c == 3 else torch.cat((p0, x0), 1)
        p1, x1, o1 = self.enc1([p0, x0, o0])
        p2, x2, o2 = self.enc2([p1, x1, o1])
        p3, x3, o3 = self.enc3([p2, x2, o2])
        p4, x4, o4 = self.enc4([p3, x3, o3])
        p5, x5, o5 = self.enc5([p4, x4, o4])
        x5 = self.dec5[1:]([p5, self.dec5[0]([p5, x5, o5]), o5])[1]
        x4 = self.dec4[1:]([p4, self.dec4[0]([p4, x4, o4], [p5, x5, o5]), o4])[1]
        x3 = self.dec3[1:]([p3, self.dec3[0]([p3, x3, o3], [p4, x4, o4]), o3])[1]
        x2 = self.dec2[1:]([p2, self.dec2[0]([p2, x2, o2], [p3, x3, o3]), o2])[1]
        x1 = self.dec1[1:]([p1, self.dec1[0]([p1, x1, o1], [p2, x2, o2]), o1])[1]
        x = self.cls(x1)

        b = o0.shape[0]
        n,c = x.shape
        x = x.reshape(b,n//b,c)
        return x
```
then, I try to make the input as one but not tuple as shown below:
the evaluate code 
```python
input = Variable(input_xyz,requires_grad=True)
logits = model(input)
loss = criterion(logits, target.squeeze())
loss.backward()
```
the forward code of origin code:
```python
def forward(self, pc):
        n,c = pc.shape
        p0, x0, o0 = pc[...,:3].clone().contiguous(),pc,torch.IntTensor([n]).cuda() # (n, 3), (n, c), (b)
        # x0 = p0 if self.c == 3 else torch.cat((p0, x0), 1)
        p1, x1, o1 = self.enc1([p0, x0, o0])
        p2, x2, o2 = self.enc2([p1, x1, o1])
        p3, x3, o3 = self.enc3([p2, x2, o2])
        p4, x4, o4 = self.enc4([p3, x3, o3])
        p5, x5, o5 = self.enc5([p4, x4, o4])
        x5 = self.dec5[1:]([p5, self.dec5[0]([p5, x5, o5]), o5])[1]
        x4 = self.dec4[1:]([p4, self.dec4[0]([p4, x4, o4], [p5, x5, o5]), o4])[1]
        x3 = self.dec3[1:]([p3, self.dec3[0]([p3, x3, o3], [p4, x4, o4]), o3])[1]
        x2 = self.dec2[1:]([p2, self.dec2[0]([p2, x2, o2], [p3, x3, o3]), o2])[1]
        x1 = self.dec1[1:]([p1, self.dec1[0]([p1, x1, o1], [p2, x2, o2]), o1])[1]
        x = self.cls(x1)

        # b = o0.shape[0]
        # n,c = x.shape
        # x = x.reshape(b,n//b,c)
        return x
```
But they both throw the error and I don't know how to solve it.At the same time,I didn't find a similary problem as me on StackOverFlow/GitHub.
```
Exception has occurred: NotImplementedError       (note: full exception trace is shown but execution is paused at: _run_module_as_main)
You must implement the backward function for custom autograd.Function.
  File "/home/gz/anaconda3/envs/pc/lib/python3.8/site-packages/torch/autograd/function.py", line 201, in backward
    raise NotImplementedError("You must implement the backward function for custom"
  File "/home/gz/anaconda3/envs/pc/lib/python3.8/site-packages/torch/autograd/function.py", line 89, in apply
    return self._forward_cls.backward(self, *args)  # type: ignore
  File "/home/gz/anaconda3/envs/pc/lib/python3.8/site-packages/torch/autograd/__init__.py", line 145, in backward
    Variable._execution_engine.run_backward(
  File "/home/gz/anaconda3/envs/pc/lib/python3.8/site-packages/torch/tensor.py", line 245, in backward
    torch.autograd.backward(self, gradient, retain_graph, create_graph, inputs=inputs)
  File "/home/gz/Desktop/3DCVcode/segmentation/PointNet12KiTTI/adv_generate.py", line 188, in generate_adv_sample
    loss.backward()
  File "/home/gz/Desktop/3DCVcode/segmentation/PointNet12KiTTI/adv_generate.py", line 239, in main
    adv_sample = generate_adv_sample(items ,alpha = 0.016,method = adv_method,model = model,criterion = criterion, args = args)
  File "/home/gz/Desktop/3DCVcode/segmentation/PointNet12KiTTI/adv_generate.py", line 246, in <module>
    main()
  File "/home/gz/anaconda3/envs/pc/lib/python3.8/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "/home/gz/anaconda3/envs/pc/lib/python3.8/runpy.py", line 97, in _run_module_code
    _run_code(code, mod_globals, init_globals,
  File "/home/gz/anaconda3/envs/pc/lib/python3.8/runpy.py", line 265, in run_path
    return _run_module_code(code, init_globals, run_name,
  File "/home/gz/anaconda3/envs/pc/lib/python3.8/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "/home/gz/anaconda3/envs/pc/lib/python3.8/runpy.py", line 194, in _run_module_as_main (Current frame)
    return _run_code(code, main_globals, None,
```

### Versions

GPU RTX 3090
CUDA 11.1
Ubuntu20.04
PyTorch 1.8.1

**Comment by bdhirsh** [comment]
Hi, We use github issues only for bug reports or feature requests. Please use the forum to ask questions: https://discuss.pytorch.org/
